woocommerce/woocommerce-ios
# Product Preview: Add "Preview" button to product details

Closes: https://github.com/woocommerce/woocommerce-ios/issues/7944

## Description

This PR adds button on product details to show a preview. Button should appear:
- for new product flow (when it's possible to create a draft, but not for a completely blank product)
- for existing drafts (unchanged and modified).

It triggers draft saving (if needed) before opening webview with preview URL (unauthenticated at this stage).
The whole new feature is hidden behind new feature flag.

## Testing

1. Open products tab.
2. Open any existing published product.
    1. Confirm that there is no "Preview" button in navbar.
    2. Update any product property.
    3. Confirm that "Save" button appears, but there is still no "Preview" button.
3. Start the flow to create new product ("+" button in navbar on product list)
    1. Edit the title (or anything else) to have non-blank product.
    2. Confirm that "Preview" button appears in navbar.
    3. Tap it and check that draft is saved before WebView opens (ignore web content).
4. Open any existing draft product (you can continue with product from last step).
    1. Confirm that "Preview" button appears in navbar.
    4. Tap it and check that WebView immediately opens (ignore web content).
    5. Update any product property.
    6. Confirm that "Preview" button still available and tap it.
    7. Confirm that draft is saved before WebView opens.

## Screenshots

before|after
--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/198312098-b96ab486-f091-491e-9ca0-f6a9ac181023.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/198312107-a188c859-1c4f-4585-b083-4dfce05e9e9c.png)

---
